### DISCUSSION TITLE
UserScript_obsPrecip_obsOnly_Hovmoeller

### DISCUSSION THREAD

**Discussion by mgehne** [dtcenter/METplus#1645] [Configuration] [answered]
Hi,

I'm using the Hovmoeller user script on HERA. My final goal is to run that on multiple input files and generate a plot for each input file with different labels etc. 
I can run the default case using run_metplus.py -c UserScript_obsPrecip_obsOnly_Hovmoeller.conf with the default user script and yaml file. I can also run it on a different netcdf file and I adjusted the yaml file to change the plot name and title. When I did that I noticed that any changes to the title, width or height of the plot did not seem to get used. Is it possible that the default values for width, height and title from the yaml file get overwritten somewhere?

Maria

Below is the UserScript_obsPrecip_obsOnly_Hovmoeller.conf I'm using which points to the modified yaml file.

```
[config]

PROCESS_LIST = UserScript

LOOP_BY = REALTIME
VALID_TIME_FMT = %Y
VALID_BEG = 2014

USER_SCRIPT_RUNTIME_FREQ = RUN_ONCE 

USER_SCRIPT_COMMAND = {PARM_BASE}/use_cases/model_applications/s2s/UserScript_obsPrecip_obsOnly_Hovmoeller/hovmoeller_diagram.py


[user_env_vars]

LOG_FILE = "Hovmoeller_diagram.log"

LOG_LEVEL = "INFO"

# obs precip and  yaml file
INPUT_FILE_NAME = {INPUT_BASE}/model_applications/s2s/UserScript_obsPrecip_obsOnly_Hovmoeller/precip.erai.sfc.1p0.2x.2014-2016.nc
YAML_CONFIG_NAME = {OUTPUT_BASE}/Hovmoeller/UserScript_obsPrecip_obsOnly_Hovmoeller/hovmoeller.yaml

# UFS P5 precip and yaml file
# INPUT_FILE_NAME = {OUTPUT_BASE}/Hovmoeller/data/prate_avg_ufs_p5_2014040100.nc
# YAML_CONFIG_NAME = {OUTPUT_BASE}/Hovmoeller/UserScript_obsPrecip_obsOnly_Hovmoeller/hovmoeller_ufs_p5.yaml

# UFS P7 precip and yaml file
# INPUT_FILE_NAME = {OUTPUT_BASE}/Hovmoeller/data/prate_avg_ufs_p7_2014040100.nc
# YAML_CONFIG_NAME = {OUTPUT_BASE}/Hovmoeller/UserScript_obsPrecip_obsOnly_Hovmoeller/hovmoeller_ufs_p7.yaml

METPLOTPY_BASE = {METPLUS_BASE}/../../METplotpy/METplotpy-1.1.0
OUTPUT_DIR = {OUTPUT_BASE}/plots
```
 
And the modified yaml file is this:
```
height: 1200
width: 800
title: ERAI Precipitation

font_size: 20

xaxis_title: Longitude
yaxis_title: Time

date_start: 2014-04-01
date_end: 2014-04-30

lat_max: 5
lat_min: -5

var_name: precip
var_units: mm / h
unit_conversion: 167

contour_min: 0.2
contour_max: 1.6
contour_del: 0.2
colorscale: 'BuPu'
plot_filename: 
- !ENV '${OUTPUT_DIR}/erai_precip_20140401-20140430.png'
```

**Comment by jprestop** [comment]
Hi @mgehne.  Thank you for your question.  I tried to get on Hera this morning to take a look at what might be going on, before realizing that Hera is down this morning.  I am tagging @fisherhucar and @georgemccabe who are more familiar with this use case and may be able to identify what is going on without having access to the files. @fisherhucar and/or @georgemccabe can you provide any guidance to @mgehne?

**Reply by mgehne** [reply]
Thanks @jprestop. They are doing the monthly maintenance on Hera today, I believe.

**Comment by jprestop** [comment]
Hi @mgehne. Now that Hera is back up, I first tried to run the default case that you were able to run successfully.  Unfortunately, I was not as fortunate and am having trouble running the default use case.  In our team Slack channel, I have asked @fisherhucar and @georgemccabe for some assistance in getting the default use case to work so that I can try to help with the modified yaml file.  I will follow up once I have more information.  Thank you for your patience.

**Reply by mgehne** [reply]
I had to change the path to the METplotpy in the .conf file to METPLOTPY_BASE = {METPLUS_BASE}/../../METplotpy/METplotpy-1.1.0
OUTPUT_DIR = {OUTPUT_BASE}/plots
It seemed to be looking for that in the wrong location before.

**Reply by jprestop** [reply]
Thank you so much, @mgehne!  That was very helpful and after that change, the default use case ran successfully for me as well.  I will follow up shortly with more information.

**Comment by jprestop** [comment, marked as answer]
After running with the changes from the yaml file listed above and taking a look at the hovmoeller_diagram.py file, it looks like only the following variables are being read in from the yaml file:

- date_start
- date_end
- var_name
- var_units
- unit_conversion
- plot_filename

That leaves the following variables in the yaml file that do not appear to be being read:

- height
- width
- title
- font_size
- xaxis_title
- yaxis_title
- lat_max
- lat_min
- contour_min
- contour_max
- contour_del
- colorscale

I do not know what the code *should* be doing or what is expected for this case, so I will need to defer to the team to find out what should be happening.

@fisherhucar, @bikegeek, and @georgemccabe could you please provide some guidance for this situation?  Also, does the use case .conf file need any changes regarding the change to the METPLOTPY_BASE value that was necessary as stated in the comments above, or is this a specific to Hera problem?  Thanks in advance for any guidance you can give.

**Reply by mgehne** [reply]
I think this is what I was looking for! Thanks for your help with this. It looked like those should be configurable and I wanted to make sure I wasn't missing anything.

**Comment by bikegeek** [comment]
Just a "heads-up", I've made modifications to the hovmoeller plot code
(METplotpy issue #213) to make it configurable.  As it is currently
written, the user cannot specify the output location of the plot (when
created from the command line).  This will result in changes to the code in
the METplus use case, but these changes haven't been merged into the
develop branch pending more testing before submitting a PR.  The
modifications allow the user to override the default values for height,
width, title, etc. set in the
METplotpy/metplotpy/plots/config/default_hovmoeller.yaml file.

I don't know what the expected behavior is when using the METplus use case
code, I will defer to @Hank Fisher ***@***.***> to answer those
questions.
---------------
Minna Win
Pronouns: she/her
National Center for Atmospheric Research
Developmental Testbed Center
Phone: 303-497-8423
Fax:   303-497-8401
---------------
